iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/functional_interface.inl

```diff
@@ -37,8 +43,12 @@ inline void Expect<Derived>::expect(const char* const msg) const noexcept
 }
 
 template <typename Derived, typename ValueType>
-inline ValueType& ExpectWithValue<Derived, ValueType>::expect(const char* const msg) & noexcept
+template <typename StringType>
+inline ValueType& ExpectWithValue<Derived, ValueType>::expect(const StringType& msg) & noexcept
 {
+    static_assert(is_char_array<StringType>::value || is_cxx_string<StringType>::value,
```

Comment:
Why didn't you add the `static_assert` to the other methods? E.g. `ExpectWithValue<Derived, ValueType>::expect(const StringType& msg)`

Reply:
Since the other methods are forwarding the arguments to this method to reduce the code duplication and implement this only once.

So it is always verified and adding this to all methods would just increase the code duplication.